network/test/epochtransition_test.go

```diff
@@ -51,35 +51,75 @@ type testNode struct {
 }
 
 // testNodeList is a list of test node and has functions to retrieve the different elements of the test nodes
-type testNodeList []testNode
+type testNodeList struct {
+	sync.RWMutex
+	nodes []testNode
```

Comment:
the nodes list is updated by adding and removing nodes. The nodes network layer pull this list async when the `suite.signalIdentityChanged()` is called. But because it wasn't under a mutex earlier, nodes sometimes continued to run with the old list and a new node which was added by the test remained unconnected.